pkg/ruler/ruler.go

```diff
@@ -531,9 +539,11 @@ func (r *Ruler) newManager(ctx context.Context, userID string) (*promRules.Manag
 		return nil, err
 	}
 
-	// Wrap registerer with userID and cortex_ prefix
-	reg := prometheus.WrapRegistererWith(prometheus.Labels{"user": userID}, r.registry)
-	reg = prometheus.WrapRegistererWithPrefix("cortex_", reg)
+	// Create a new Prometheus registry and register it within
+	// our metrics struct for the provided user.
+	reg := prometheus.NewRegistry()
+	r.userManagerMetrics.AddUserRegistry(userID, reg)
```

Comment:
Sadly, that is the case. I'm not sure if it's an issue since it is related to deleting/recreating the manager. Since it's a new instantiation of the manager, the reset may be appropriate.

WDYT?

Reply:
I think full reset is fine, and since ruler metrics export *ALL* metrics per user, I think that should actually work. 

(In blocks storage we sometimes sum metrics from different users, and there it's not a *full* reset, but partial, which I believe would confuse PromQL.)

Reply:
👍